app/mailers/user_mailer.rb

```diff
@@ -422,6 +400,17 @@ def open_project_headers(hash)
 # Unfortunately, this results in changes on the interceptor classes during development mode
 # not being reflected until a server restart.
 
+class PlainTextInterceptor
+  def self.delivering_email(message)
+    if Setting.plain_text_mail?
+      # HACK: this is a workaround to ensure @html_part ivar is set correctly
+      # on the Mail::Message.
+      message.html_part = message.html_part
+      message.html_part = nil
```

Comment:
Can't we directly specify the templates to be rendered? 

> If you want more flexibility you can also pass a block and render specific templates or even render inline or text without using a template file
[[Action Mailer Basics - 2.4 Mailer Views](http://guides.rubyonrails.org/action_mailer_basics.html)]

There is probably a more elegant approach but naively we could specify

```
mail(to: ..., subject: ...) do |format|
 format.html unless Setting.plain_text_mail?
 format.text
end
```

in each mailer method, right?